File: nextapp/scratch_pdf.js
const fs = require('fs');
const path = require('path');

const pdfPath = path.resolve('c:/xampp/htdocs/cambridge_website/cambridge_bronchure/Cambridge Education Consultants (Broucher).pdf');

if (!fs.existsSync(pdfPath)) {
  console.log("PDF not found at:", pdfPath);
  process.exit(1);
}

console.log("Scanning PDF binary for keywords...");
const buffer = fs.readFileSync(pdfPath);
const contentString = buffer.toString('binary'); // convert to binary string

const countries = [
  'Tajikistan', 'Kazakhstan', 'Kyrgyzstan', 'Georgia', 'Bangladesh', 'Uzbekistan', 
  'Iran', 'Russia', 'Philippines', 'China', 'Ukraine', 'Armenia', 'Nepal',
  'USA', 'Canada', 'United Kingdom', 'UK', 'Ireland', 'Germany', 'France', 
  'Italy', 'Spain', 'Poland', 'Australia', 'New Zealand', 'Malaysia', 'Singapore'
];

const results = {};
countries.forEach(country => {
  // Simple regex search in binary content
  const regex = new RegExp(country, 'gi');
  const matches = contentString.match(regex);
  results[country] = matches ? matches.length : 0;
});

console.log("Country mention counts in PDF:");
console.log(JSON.stringify(results, null, 2));

File: nextapp/scratch_pdf.py
import sys
import os

pdf_path = r"c:\xampp\htdocs\cambridge_website\cambridge_bronchure\Cambridge Education Consultants (Broucher).pdf"

# Try pypdf or PyPDF2
try:
    import pypdf
    reader = pypdf.PdfReader(pdf_path)
    text = ""
    for page in reader.pages[:10]: # Read first 10 pages for overview
        text += page.extract_text() or ""
    print("Extracted from first 10 pages:")
    print(text[:1000])
except Exception as e:
    print("pypdf error:", e)
    try:
        import PyPDF2
        reader = PyPDF2.PdfReader(pdf_path)
        text = ""
        for page in reader.pages[:10]:
            text += page.extract_text() or ""
        print("Extracted from first 10 pages (PyPDF2):")
        print(text[:1000])
    except Exception as e2:
        print("PyPDF2 error:", e2)
